watertap/core/plugins/tests/test_solvers.py

```diff
@@ -0,0 +1,148 @@
+        assert m.c in cons_with_sf
+        assert m.b.c[1] in cons_with_sf
+        assert m.b.c[2] in cons_with_sf
+        assert m.b.d in cons_with_sf
+
+        assert s._model is m
+
+    @pytest.mark.unit
+    def test_postsolve_unscaled_constraints_cleanup(self, m, s):
+        assert hasattr(s, '_postsolve')
+        # for the Pyomo implementation
+        try:
+            s._postsolve()
+        except AttributeError:
+            pass
+
+        assert not hasattr(s, '_scaling_cache')
+        
+        cons_with_sf = list(constraints_with_scale_factor_generator(m))
+        assert cons_with_sf == [(m.b.d, 1e6)]
+
+        assert not hasattr(s, '_model')
+
+    @pytest.mark.unit
+    def test_option_absorption(self, m, s):
+        s.options['ignore_variable_scaling'] = True
+        results = s.solve(m, tee=True)
+        pyo.assert_optimal_termination(results)
+        del s.options['ignore_variable_scaling']
+
+    @pytest.mark.unit
+    def test_get_option(self, s):
+        s.options['ignore_constraint_scaling'] = True
+
+        assert s._get_option('ignore_constraint_scaling', False) is True
+        assert 'ignore_constraint_scaling' not in s.options
+        assert s._get_option('ignore_constraint_scaling', False) is False
+
+    @pytest.mark.unit
+    def test_presolve_passthrough(self, m, s):
+        s.options["nlp_scaling_method"] = "gradient-based"
+        s._presolve(m, tee=True)
+        assert not hasattr(s, "_scaling_cache")
+        s.options["nlp_scaling_method"] = "user-scaling"
+
+    @pytest.mark.unit
+    def test_passthrough_positive(self, m, s):
+        s.options["nlp_scaling_method"] = "gradient-based"
+        pyo.assert_optimal_termination(s.solve(m, tee=True))
+        del s.options["nlp_scaling_method"]
+
+    @pytest.mark.unit
+    def test_passthrough_negative(self, m, s):
+        s.options["nlp_scaling_method"] = "gradient-based"
+        s.options["ignore_variable_scaling"] = True
+        with pytest.raises(ApplicationError):
+            pyo.assert_optimal_termination(s.solve(m, tee=True))
+        del s.options["nlp_scaling_method"]
+        del s.options['ignore_variable_scaling']
+
+    @pytest.mark.unit
+    def test_presolve_incorrect_number_of_arguments(self, m, s):
+        with pytest.raises(TypeError):
+            s.solve()
+        with pytest.raises(TypeError):
+            s.solve(m, m)
+
+    @pytest.mark.unit
+    def test_presolve_incorrect_argument_type(self, s):
+        with pytest.raises(TypeError):
+            s.solve('abc')
+
+    @pytest.mark.unit
+    def test_presolve_AMPL_evaluation_error(self, m, s):
+        m.a.value = 0
+        with pytest.raises(RuntimeError):
+            s.solve(m)
+        m.a.value = 1
+
+    @pytest.mark.unit
+    def test_presolve_ignore_AMPL_evaluation_error(self, m, s):
+        m.a.value = 0
+        s.options["halt_on_ampl_error"] = "no"
+        s._presolve(m)
```

Comment:
I'm guessing the test passes because all that comes of this is a printed statement, correct?